src/Module/ValidateCartBookingHandler.php

```diff
@@ -0,0 +1,246 @@
+     *
+     * @param EDD_Cart                                      $eddCart          The EDD cart instance.
+     * @param TransitionerInterface                         $transitioner     The booking transitioner.
+     * @param SelectCapableInterface                        $bookingsSelectRm The bookings SELECT resource model.
+     * @param array|stdClass|ArrayAccess|ContainerInterface $cartItemConfig   The cart item data config.
+     * @param object                                        $exprBuilder      The expression builder.
+     */
+    public function __construct(
+        EDD_Cart $eddCart,
+        TransitionerInterface $transitioner,
+        SelectCapableInterface $bookingsSelectRm,
+        $exprBuilder,
+        $cartItemConfig
+    ) {
+        $this->eddCart          = $eddCart;
+        $this->transitioner     = $transitioner;
+        $this->bookingsSelectRm = $bookingsSelectRm;
+        $this->exprBuilder      = $exprBuilder;
+        $this->cartItemConfig   = $cartItemConfig;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function __invoke()
+    {
+        $items = $this->eddCart->get_contents();
+
+        // Get the cart item data keys from the cart item config
+        $dataKey       = $this->_containerGetPath($this->cartItemConfig, ['data', 'key']);
+        $eddBkKey      = $this->_containerGetPath($this->cartItemConfig, ['data', 'eddbk_key']);
+        $bookingIdKey  = $this->_containerGetPath($this->cartItemConfig, ['data', 'booking_id_key']);
+        $bookingIdPath = [$dataKey, $eddBkKey, $bookingIdKey];
+
+        // Alias expression builder
+        $b = $this->exprBuilder;
+
+        foreach ($items as $_item) {
+            try {
+                // Get the booking ID
+                $bookingId = $this->_containerGetPath($_item, $bookingIdPath);
+            } catch (NotFoundExceptionInterface $exception) {
+                continue; // If ID not found, cart item does not represent a booking
+            }
+
+            // Get the booking that matches the ID
+            $bookings  = $this->bookingsSelectRm->select(
+                $b->eq(
+                    $b->ef('booking', 'id'),
+                    $b->lit($bookingId)
+                )
+            );
+
+            // Check that only 1 booking was retrieved
+            if ($this->_countIterable($bookings) !== 1) {
+                \edd_set_error('eddbk_invalid_booking_id', $this->__('A cart item has an invalid booking ID.'));
+            }
+
+            // Get the booking
+            $booking = reset($bookings);
+
+            try {
+                // Validate it
+                $this->_validateBooking($booking);
+            } catch (ValidationFailedExceptionInterface $exception) {
+                // Register all validation errors as EDD checkout errors
+                foreach ($exception->getValidationErrors() as $key => $error) {
+                    \edd_set_error(
```

Comment:
Tired of typing. Guess what the problem is.

Reply:
It's on line 196? I'll add a new line so that it will be on 197 😆 🤣